### PR TITLE
ScalarRenderer: add dummy images for ranks with no data

### PR DESCRIPTION
**Pull request by nicolemarsaglia**
Assumes rank 0 has data and passes the necessary info for dummy images on ranks with no data.

### CODE REVIEW

**Comment by nicolemarsaglia** [comment]
I don't think any of these build errors are from my code. One is cuda + contour tree, and the other two are spack/fetching and loading containers.

File: src/vtkh/rendering/ScalarRenderer.cpp [hunk 1]
```diff
@@ -137,23 +139,56 @@ ScalarRenderer::DoExecute()
...
+      bounds[3] = pimage->m_bounds.Y.Max;
+      bounds[4] = pimage->m_bounds.Z.Min;
+      bounds[5] = pimage->m_bounds.Z.Max;
+      delete pimage;
+    }
+  }
+
+  //Assume rank 0 has data, pass details to ranks with empty domains (no data).
+#ifdef VTKH_PARALLEL
+  MPI_Bcast(bounds, 6, MPI_FLOAT, 0, MPI_COMM_WORLD);
```
Comment (mclarsen) [inline src/vtkh/rendering/ScalarRenderer.cpp]: There is a method for getting MPI comm.

```
MPI_Comm mpi_comm = MPI_Comm_f2c(vtkh::GetMPICommHandle());
```
This is inside vtkh.hpp i think.

File: src/vtkh/rendering/ScalarRenderer.cpp [hunk 2]
```diff
@@ -137,23 +139,56 @@ ScalarRenderer::DoExecute()
...
+  MPI_Bcast(&max_p, 1, MPI_INT, 0, MPI_COMM_WORLD);
```
Comment (mclarsen) [inline src/vtkh/rendering/ScalarRenderer.cpp]: Also, I am not sure that rank 0 has any promise of having any data. You will probably need to figure out who has data and make that the root rank.

**Review by mclarsen — CHANGES_REQUESTED** [review]

File: src/vtkh/rendering/ScalarRenderer.cpp (hunk 1 above)
Comment (nicolemarsaglia) [reply src/vtkh/rendering/ScalarRenderer.cpp]: got it! I always forget about the mpi wrappers.

File: src/vtkh/rendering/ScalarRenderer.cpp (hunk 2 above)
Comment (nicolemarsaglia) [reply src/vtkh/rendering/ScalarRenderer.cpp]: Can this be a future todo? I'm mainly trying to get some basic parallel runs done for the milestone. I'll probably have to fix this once I connect to exawind.

Comment (mclarsen) [reply src/vtkh/rendering/ScalarRenderer.cpp]: Ok. I guess its no worse than what is was.

**Pull request merged**